### PR TITLE
fix: query string is incorrectly serialized for primitive parameters (#103)

### PR DESCRIPTION
**Pull request by kokovtsev**

### CODE REVIEW

File: src/language/typescript/2.0/serializers/operation-object.ts [hunk 1]
```diff
@@ -411,8 +411,8 @@ export const serializeQueryParameterObject = (
 		case 'number':
 		case 'boolean': {
 			const f = serializedFragment(
-				`value => encodeURIComponent('${parameter.name}') + '=' + encodeURIComponent(value)`,
-				[],
+				`value => some(encodeURIComponent('${parameter.name}') + '=' + encodeURIComponent(value))`,
```
Comment (raveclassic) [inline src/language/typescript/2.0/serializers/operation-object.ts:414]: I see that you add `some` in all `switch` branches, would it make more sense to keep this function simpler and operate on simple types and wrap its result in `some` on the call site?

File: src/language/typescript/3.0/serializers/parameter-object.ts [hunk 2]
```diff
@@ -121,26 +121,32 @@ const getFn = (
 	if (PrimitiveSchemaObjectCodec.is(schema)) {
 		return right(
 			serializedFragment(
-				`value => serializePrimitiveParameter('${style}', '${parameter.name}', value)`,
-				[serializedDependency('serializePrimitiveParameter', pathToUtils)],
+				`value => option.fromEither(serializePrimitiveParameter('${style}', '${parameter.name}', value))`,
+				[
+					serializedDependency('option', 'fp-ts'),
```
Comment (raveclassic) [inline src/language/typescript/3.0/serializers/parameter-object.ts:126]: although it's ok to import modules from fp-ts index in the end project, here in the lib we'd like to minimise imports, so could you import `fromEither` directly from `fp-ts/lib/Option` here?

File: src/language/typescript/3.0/serializers/parameter-object.ts [hunk 3]
```diff
@@ -121,26 +121,32 @@ const getFn = (
...
 				[],
 			),
 		);
 	}
 	if (ArraySchemaObjectCodec.is(schema)) {
 		return right(
 			serializedFragment(
-				`value => serializeArrayParameter('${style}', '${parameter.name}', value, ${explode})`,
-				[serializedDependency('serializeArrayParameter', pathToUtils)],
+				`value => option.fromEither(serializeArrayParameter('${style}', '${parameter.name}', value, ${explode}))`,
```
Comment (raveclassic) [inline src/language/typescript/3.0/serializers/parameter-object.ts:136]: idem

File: src/language/typescript/common/data/serialized-fragment.ts [hunk 4]
```diff
@@ -136,12 +139,19 @@ export const getSerializedOptionCallFragment = (
...
 							option.fromNullable,
-							option.map(${fn}),
+							option.chain(${fn}),
 						)`,
 					[serializedDependency('option', 'fp-ts'), serializedDependency('pipe', 'fp-ts/lib/pipeable')],
 					[],
 			  )
-			: serializedFragment(`some((${fn})(${a}))`, [serializedDependency('some', 'fp-ts/lib/Option')], []),
+			: serializedFragment(
+					`pipe(
```
Comment (raveclassic) [inline src/language/typescript/common/data/serialized-fragment.ts:148]: minor: is it really necessary to change direct function invocation to pipe?

Comment (kokovtsev) [reply src/language/typescript/common/data/serialized-fragment.ts:148]: It's just a matter of readability; direct invocation turns out less readable even after applying prettier. Shall I switch it back to the direct call?
image

Comment (raveclassic) [reply src/language/typescript/common/data/serialized-fragment.ts:148]: let's skip

File: src/language/typescript/3.0/serializers/parameter-object.ts (hunk 2 above)
Comment (kokovtsev) [reply src/language/typescript/3.0/serializers/parameter-object.ts:126]: Aren't we going to have collisions if some other serializedFragment imports fromEither from another module?

Comment (raveclassic) [reply src/language/typescript/3.0/serializers/parameter-object.ts:126]: The same is for `option`/`either` etc from `fp-ts`. We don't have general solution at the moment.

**Pull request merged**